fix: Add clarification to Quarkus chapter (#876)
vaadin/docs: articles/flow/integrations/quarkus.asciidoc

This starter is also available with Gradle configuration in the https://github.com/vaadin/base-starter-flow-quarkus/tree/gradle[Gradle] branch.

[[quarkus.dependencies]]
== Adding Dependencies

> https://github.com/vaadin/base-starter-flow-quarkus/[Quarkus base starter] has all the needed dependencies to start developing your application.

Reviewer: This line is extra and the `Adding Dependencies` should be renamed to `Manual Setup` or something as just before this "chapter" there is the line `It is a project template with the necessary configuration and dependencies included for starting to build your own application.` saying exactly this. It shouldn't need duplicating.

Author: done

quarkus.index-dependency.<name>.classifier=(this one is optional)
----

The `<name>` string here is used to link the `group-id`, `artifact-id` and `classifier` entries in one logical block.
It should be the same for those three entries and be any string literal.

== Running the Application in Development Mode

After making <<quarkus.setup>> steps, the Quarkus application can be started in development mode using the `quarkus:dev` goal in Maven.

[source,terminal]
----
$ mvn package quarkus:dev
----

The application is then available at http://localhost:8080/[+localhost:8080+] in the browser.

== Running the Application in Production Mode

Quarkus base starter already includes the needed Maven configuration for running the application in production mode.
If you use a project not based on the starter, it needs the configuration described in https://vaadin.com/docs/latest/flow/guide/production/#enabling-the-production-mode[Enabling the Production Mode].

Run the following commands to start the application:

> ```
> mvn package -Pproduction
> java -jar target/quarkus-app/quarkus-run.jar
> ```

Reviewer: To keep the same format as used before, should we have this as
```suggestion
[source,terminal]
----
$ mvn package -Pproduction
$ java -jar target/quarkus-app/quarkus-run.jar
----
```

Reviewer 2: I think we shouldn't use the `$` symbol in terminal commands. Doesn't that make it difficult to copy paste?

Reviewer: That would be good too. I expect the whole documentation should be looked through to not have them even for single line commands.

Reviewer 2: I’ll unify these in the latest branch.

Reviewer 2: Updated everything else except this Quarkus article.

quarkus.index-dependency.<name>.classifier=(this one is optional)
----

The `<name>` string here is used to link the `group-id`, `artifact-id` and `classifier` entries in one logical block.
It should be the same for those three entries and be any string literal.

== Running the Application in Development Mode

After making <<quarkus.setup>> steps, the Quarkus application can be started in development mode using the `quarkus:dev` goal in Maven.

[source,terminal]
----

> $ mvn package quarkus:dev

Reviewer 2: ```suggestion
mvn package quarkus:dev
```

----

The application is then available at http://localhost:8080/[+localhost:8080+] in the browser.

== Running the Application in Production Mode

Quarkus base starter already includes the needed Maven configuration for running the application in production mode.

> If you use a project not based on the starter, it needs the configuration described in https://vaadin.com/docs/latest/flow/guide/production/#enabling-the-production-mode[Enabling the Production Mode].

Reviewer 2: ```suggestion
If you use a project not based on the starter, it needs the configuration described in <<{articles}/flow/guide/production#enabling-the-production-mode, Enabling the Production Mode>>.
```

quarkus.index-dependency.<name>.classifier=(this one is optional)
----

The `<name>` string here is used to link the `group-id`, `artifact-id` and `classifier` entries in one logical block.
It should be the same for those three entries and be any string literal.

== Running the Application in Development Mode

After making <<quarkus.dependencies>> steps, the Quarkus application can be started in development mode using the `quarkus:dev` goal in Maven.

> [source,bash]

Reviewer 2: Let’s use "terminal" as the general term for command line snippets.

```suggestion
[source,terminal]
```

Author: done

== Running the Application in Production Mode

If you want to run your application locally in production mode, add the following `production` profile to your project:

.`pom.xml`
[source,xml]
----
<profile>
    <id>production</id>
    <properties>
        <vaadin.productionMode>true</vaadin.productionMode>
    </properties>

    <dependencies>
        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>flow-server-production-mode</artifactId>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>com.vaadin</groupId>
                <artifactId>vaadin-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <goals>
                            <goal>build-frontend</goal>
                        </goals>
                        <phase>compile</phase>
                    </execution>
                </executions>
                <configuration>
                    <productionMode>true</productionMode>
                </configuration>
            </plugin>
        </plugins>
    </build>
</profile>

> ----

Reviewer: This is the default production mode profile and should just link to https://vaadin.com/docs/latest/flow/guide/production/#enabling-the-production-mode with the addition of if you didn't start from the starter.

Author: done

== Running the Application in Production Mode

> Quarkus base starter already includes the needed maven configuration for running application in production mode.

Reviewer 2: ```suggestion
Quarkus base starter already includes the needed Maven configuration for running the application in production mode.
```
